client/scripts/views/pages/view_proposal/index.ts

```diff
@@ -505,25 +506,31 @@ const ViewProposalPage: m.Component<{
     }
 
     // load proposal
-    let proposal: AnyProposal;
-    try {
-      proposal = idToProposal(proposalType, proposalId);
-    } catch (e) {
-      // proposal might be loading, if it's not an offchain thread
-      if (proposalType !== ProposalType.OffchainThread) {
-        if (!app.chain.loaded) return m(PageLoading, { narrow: true, showNewProposalButton: true });
-
-        // check if module is still initializing
-        const c = proposalSlugToClass().get(proposalType) as ProposalModule<any, any, any>;
-        if (!c.disabled && !c.initialized) {
-          if (!c.initializing) loadCmd(proposalType);
+    if (!vnode.state.proposal) {
+      try {
+        vnode.state.proposal = idToProposal(proposalType, proposalId);
+      } catch (e) {
+        // proposal might be loading, if it's not an offchain thread
+        if (proposalType === ProposalType.OffchainThread) {
+          app.threads.fetchThread(Number(proposalId)).then((res) => {
```

Comment:
We should make sure this doesn't run multiple times, since it's in the `view` code for the component and view code is rerun every time the code rerenders.